### ISSUE TITLE
Running LDpred2 without internet server access 

### ISSUE THREAD

**Issue by uki-uiu** [privefl/bigsnpr#474]
i am trying to run LDpred2 using hg38 genome-wide data offline however the script does not allow me to select the latest genetic map coordinates for this purpose; 
`
POS2 <- snp_asGeneticPos(CHR, POS, dir = "/filepath/genetic_map_hg38_withX.txt.gz").`

This step only works with internet access and repeatedly tries to access the OMNI files from the hg19 build (chromosome separated). Is there a work around for this step? 

I will be using an LD reference based on the local dataset so I do not need the hapMap data but will need to calculate the correlation values from the matrix in this step.

**Comment by privefl** [comment]
You need to ask for those files to be put on your server.
This has been discussed in other issues here.
BTW, `dir` is for the directory, not the full path.

Otherwise, you can use something like 3MB window, identify nearly-independent LD blocks from that, and then re-compute all the values within the LD blocks (by using something like POS <- block_id, and size = 1e-4); this will probably give you the best LD matrix.

**Comment by privefl** [comment]
There are several issues like this here.
Basically, if you have NAs in the polygenic scores, it means you either have NAs in the effects you get from LDpred2-auto, or you have NAs in the genotype matrix that you use to compute the polygenic scores.

**Comment by privefl** [comment]
Any update on this?

**Comment by uki-uiu** [comment]
I managed to solve the issue after going through the the other issues/solutions posted. Thank you

**Comment by uki-uiu** [comment]
I ended up using the Pred_grid option and imputed any missing genotypes
`G2<-snp_fastImputeSimple(G,method="mean2")`

And to avoid the issues I was facing using the script for scoring the test-set, I extracted the SNPs from the PGS (with the best parameters after tuning with the validation cohort) created in the earlier steps
``
I then used another tool to score the participants in the test set. 

However I have an additional question; "You mentioned that "in practice we recommend to test multiple values for h2 and p"; so I assumed that we need to explore different values for these hyper-paramaters. However, when I tried other values they did not work.
I ended up using your settings for the p_seq and h2_seq from the tutorial example :

`h2_seq<- round(ldsc_h2_est *c(0.3,0.7,1,1.4),4)
(p_seq<-signif(seq_log(1e-5,1,length.out=21,2))`

I just want to confirm whether it is acceptable/recommended to use this range of parameters provided in the tutorial or should we change them?
